persist/postgres/sectors.go

```diff
@@ -189,11 +189,37 @@ func (s *Store) PinSlab(ctx context.Context, account proto.Account, nextIntegrit
 
 		// insert slab's sectors in a single batch
 		batch := &pgx.Batch{}
-		for i, sector := range slab.Sectors {
-			batch.Queue(`INSERT INTO sectors (sector_root, host_id, slab_id, slab_index, next_integrity_check) VALUES ($1, (SELECT id FROM hosts WHERE public_key = $2), $3, $4, $5)`, sqlHash256(sector.Root), sqlPublicKey(sector.HostKey), slabID, i, nextIntegrityCheck)
+		for _, sector := range slab.Sectors {
+			batch.Queue(`
+				INSERT INTO sectors (sector_root, host_id, next_integrity_check) 
+				VALUES ($1, (SELECT id FROM hosts WHERE public_key = $2), $3) 
+				ON CONFLICT (sector_root) DO UPDATE SET sector_root=EXCLUDED.sector_root 
+				RETURNING id
+			`, sqlHash256(sector.Root), sqlPublicKey(sector.HostKey), nextIntegrityCheck)
+		}
+
+		// fetch sector IDs
+		br := tx.SendBatch(ctx, batch)
+		sectorIDs := make([]int64, len(slab.Sectors))
+		for i := range slab.Sectors {
+			if err := br.QueryRow().Scan(&sectorIDs[i]); err != nil {
+				br.Close()
+				return fmt.Errorf("failed to get sector id for index %d: %w", i, err)
+			}
+		}
+		br.Close()
```

Comment:
You can simplify a bit and get rid of the `Close` calls in the loop.

```suggestion
		sectorIDs := make([]int64, len(slab.Sectors))
		for i, sector := range slab.Sectors {
			batch.Queue(`
				INSERT INTO sectors (sector_root, host_id, next_integrity_check) 
				VALUES ($1, (SELECT id FROM hosts WHERE public_key = $2), $3) 
				ON CONFLICT (sector_root) DO UPDATE SET sector_root=EXCLUDED.sector_root 
				RETURNING id
			`, sqlHash256(sector.Root), sqlPublicKey(sector.HostKey), nextIntegrityCheck).QueryRow(func(row pgx.Row) error {
				return row.Scan(&sectorIDs[i])
			})
		}

		// fetch sector IDs
		if err := tx.SendBatch(ctx, batch).Close(); err != nil {
			return fmt.Errorf("failed to insert sectors: %w", err)
		}
```